lib/web_ui/lib/src/engine/text_editing/text_editing.dart

```diff
@@ -580,6 +581,43 @@ class IOSTextEditingStrategy extends DefaultTextEditingStrategy {
     _positionInputElementTimer?.cancel();
     _positionInputElementTimer = null;
   }
+
+  /// On iOS long press works differently than a single tap.
+  ///
+  /// On a normal tap the virtual keyboard comes up and users can enter text
+  /// using the keyboard.
+  ///
+  /// The long press on the other hand focuses on the element without bringing
+  /// up the virtual keyboard. It allows the users to modify the field by using
+  /// copy/cut/select/paste etc.
+  ///
+  /// After a long press [domElement] is positioned to the correct place. If the
+  /// user later single-tap on the [domElement] the virtual keyboard will come
+  /// and might shift the page up.
+  ///
+  /// In order to prevent this shift, on a `click` event the position of the
+  /// element is again set somewhere outside of the page and
+  /// [_positionInputElementTimer] timer is restarted. The element will be
+  /// placed to its correct position after [_delayBeforePositioning].
+  void _addTapListener() {
+    _subscriptions.add(domElement.onClick.listen((_) {
+      // Check if the element is already positioned. If not this does not fall
+      // under `The user was using the long press, now they want to enter text
+      // via keyboard` journey.
+      if (_canPosition) {
+        // Re-place the element somewhere outside of the screen.
+        domElement.style.transform = 'translate(-9999px, -9999px)';
+        _canPosition = false;
+
+        // Re-configure the timer to place the element.
+        _positionInputElementTimer?.cancel();
+        _positionInputElementTimer = Timer(_delayBeforePositioning, () {
+          _canPosition = true;
+          positionElement();
+        });
```

Comment:
The exact same block of code is repeated in the onFocus listener and I'm concerned about the two getting out of sync eventually. Could we maybe move this code into a method (e.g. `schedulePositioning()`)?

Reply:
done.